api/src/main/java/gov/cms/ab2d/api/security/JwtTokenAuthenticationFilter.java

```diff
@@ -0,0 +1,98 @@
+package gov.cms.ab2d.api.security;
+
+import com.google.common.base.Strings;
+import com.okta.jwt.AccessTokenVerifier;
+import com.okta.jwt.Jwt;
+import com.okta.jwt.JwtVerificationException;
+import com.okta.jwt.JwtVerifiers;
+import gov.cms.ab2d.common.model.User;
+import gov.cms.ab2d.common.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.time.Duration;
+
+@Component
+public class JwtTokenAuthenticationFilter extends OncePerRequestFilter {
+
+    private final AccessTokenVerifier jwtVerifier;
+
+    @Autowired
+    private JwtConfig jwtConfig;
+
+    @Autowired
+    private UserService userService;
+
+    public JwtTokenAuthenticationFilter() {
+        jwtVerifier = JwtVerifiers.accessTokenVerifierBuilder()
+                .setIssuer("https://dev-418212.okta.com/oauth2/default")
+                .setAudience("api://default")                // defaults to 'api://default'
+                .setConnectionTimeout(Duration.ofSeconds(1)) // defaults to 1s
+                .setReadTimeout(Duration.ofSeconds(1))       // defaults to 1s
```

Comment:
Let's make these two configurable properties as well. 1 second is likely too aggresive; 5 seconds might be a more conservative choice.